tools/formatting/format-check.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def check(base_commit):
+    try:
+        p = subprocess.Popen(
+                [
+                    "python",
+                    os.path.join("tools", "formatting", "git-clang-format.py"),
+                    "--style=file",
+                    "--diff",
+                    "--commit",
+                    base_commit,
+                    ],
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                )
+        out, err = p.communicate()
+    except OSError as e:
+        print("{}\n\n{}".format("Failed to call git-clang-format.py", str(e)))
```

Comment:
You don't need to manually convert args of `str.format` to `str`. `format` will do it for you.
And in such prints is better to use `repr` of printing object. At least it puts the quotes around printing string.

```python
print("Failed to call git-clang-format.py\n\n{!r}".format(e))
```